linked_list/main.py

```diff
@@ -62,34 +66,106 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        # start at none
+        self.start = None
+        self.end = None
+        # set elements to empty list
+        elements = [] if not elements else elements
+        for val in elements:
+            self.append(val)
+            
 
     def __str__(self):
-        pass
-
+        return str([element.elem for element in self])
+    
     def __len__(self):
-        pass
+        return self.count()
 
     def __iter__(self):
-        pass
+        n = self.start
+        while n:
+            yield n
+            n = n.next
 
     def __getitem__(self, index):
-        pass
+       for n, node in enumerate(self): 
+           if n == index:
+               return node
 
     def __add__(self, other):
-        pass
+       l1 = [ n.elem for n in self]
+       l2 = [ n.elem for n in other]
+       l3 = l1+l2
+       return LinkedList(l3)
 
     def __iadd__(self, other):
-        pass
+        return self.__add__(other) #  x += 1  == x = x+1
 
     def __eq__(self, other):
-        pass
+        return [element.elem for element in self] == [element.elem for element in other]
 
     def append(self, elem):
-        pass
-
+        #new node
+        if self.start == None:  #is it the first node
+            self.start = Node(elem)
+            self.end = Node(elem)
+        elif self.start == self.end:
+            self.start.next = Node(elem) #fix first node when adding 2nd
+            self.end.next = self.start.next #prior node now  points to new node
+            self.end = self.end.next #new node is now end
+        else:
+            self.end.next = Node(elem) #prior node now  points to new node
+            self.end = self.end.next #new node is now end
+                
+        
     def count(self):
-        pass
-
+        c = 0
+        for e in self:
+            c +=1
+        return c
+        
     def pop(self, index=None):
-        pass
+        prior = self.start
+        if len(self)== 0: #If the list is empty, raise an error
+            raise IndexError()
+            
+        elif index == None: #If we're not passing an index
```

Comment:
You could make your this a lot shorter if you just changed the index to refer to the last list element in case it is none.
Your function will need to handle that case anyways.